streams/src/main/java/org/apache/kafka/streams/processor/internals/ThreadMetadataImpl.java

```diff
@@ -92,7 +92,7 @@ public String restoreConsumerClientId() {
     }
 
     public Set<String> producerClientIds() {
-        return producerClientIds;
+        return Collections.singleton(producerClientId);
```

Comment:
We have a general idea to refactor the metadata hierarchy: https://issues.apache.org/jira/browse/KAFKA-12370

I have add this as part of that scope to piggyback with the proposed KIP.